server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -47,6 +47,8 @@ public class RemoteFsTranslog extends Translog {
     private final FileTransferTracker fileTransferTracker;
     private volatile long maxRemoteTranslogGenerationUploaded;
 
+    private volatile long minSeqNoToKeep;
```

Comment:
The default value of 0 should be good when `minSeqNoToKeep` has not been set ever?

Reply:
Could there be a case when a replica-primary promotion has happened and this has not been initialised or primary-primary relocation (peer recovery) in which case this can be problematic?

Reply:
In worst case, the default value of 0 would prevent further truncations. There would be some assertion failures , which would be disabled in prod.